### PR TITLE
Deprecate <~> operator

### PR DESCRIPTION
**Pull request by benharsh**
DRAFT

This PR updates our modules and tests to no longer use the <~> operator, and adds relevant deprecation warnings.

Testing:
- [ ] full local
- [ ] gasnet

TODO:
- [x] investigate weird ".read()" behavior in recordio.chpl
- [ ] get feedback on stylistic changes
- [x] add deprecation test
- [x] update existing tests
- [x] Add compiler deprecation warning for user-defined <~>

### CODE REVIEW

**Comment by benharsh** [comment]
Note that this will depend on https://github.com/chapel-lang/chapel/pull/19983 being merged.

**Comment by benharsh** [comment]
@lydia-duncan : this is finally ready for review.

I've tried to make things look nice when possible, but there are a lot of ``if f.writing then f.write(....); else f.read(....);`` instances to replace ``<~>``. We should consider whether this is acceptable.

In order to preserve the current behavior on main, I've used 'readIt' in our internal/standard/package modules rather than 'read'. This is because 'readIt' will throw an EEOF error, whereas 'read' will simply return false.

This bool-returning functionality is nice in a while loop, but isn't all that helpful when trying to implement 'readThis' on a type. Checking that returned value is a bit cumbersome in that case, hence the admittedly ugly changes in this PR to use readIt.

**Comment by benharsh** [comment]
Anyways, I imagine this ultimately ties into some broader IO topic about the various read methods returning a bool. It would be nice if we could have [something like swift here](https://stackoverflow.com/questions/32390611/try-try-try-what-s-the-difference-and-when-to-use-each). Might look like ``while ! try? f.read(....) { }``

**Comment by benharsh** [comment]
@e-kayrakli : can you take a look at the prediff change to a local accessor optimization test ([0ef5dad](https://github.com/chapel-lang/chapel/pull/19985/commits/0ef5dad60b8c526e07788c066bc81a7a2711cdc0)), and tell me whether this is reasonable or not?

File: test/optimizations/autoLocalAccess/localArrays-1D-COO.good [hunk 1]
```diff
@@ -12,7 +12,7 @@ End analyzing forall (localArrays.chpl:56)
 
 Static check successful. Using localAccess (localArrays.chpl:57)
 Testing 1D COO
-2
+1 2
```
Comment (e-kayrakli) [inline test/optimizations/autoLocalAccess/localArrays-1D-COO.good:15]: While the changes look good, I can't understand the good file before your change. It looks like it should have always been printing out `1 2`. Do you have a guess why it wasn't? I don't really remember.

**Comment by benharsh** [comment]
I'm going to split this into multiple PRs as the changes replacing the IO operator are probably worth reviewing independently of deprecation.

**Pull request closed without merging**